classes/file/PKPLibraryFileManager.php

```diff
@@ -112,16 +112,47 @@ public function &copyFromTemporaryFile(&$temporaryFile, $libraryFileType)
         $libraryFileDao = DAORegistry::getDAO('LibraryFileDAO'); /** @var LibraryFileDAO $libraryFileDao */
         $libraryFile = $libraryFileDao->newDataObject();
 
+        $libraryFile = $this->assignFromTemporaryFile($temporaryFile, $libraryFileType, $libraryFile);
+        if (!$this->copyFile($temporaryFile->getFilePath(), $this->getBasePath() . $libraryFile->getServerFileName())) {
+            return false;
+        }
+
+        return $libraryFile;
+    }
+
+    /**
+     * Routine to replace a library file from a temporary file.
+     * @param $libraryFileType int LIBRARY_FILE_TYPE_...
+     * @return LibraryFile|false the updated LibraryFile, or false on error
+     */
+    function replaceFromTemporaryFile(object &$temporaryFile, int $libraryFileType, LibraryFile $libraryFile) {
```

Comment:
`object` should be `TemporaryFile` (and a `use` statement for that should be added at the top).

Reply:
Also, the reference isn't necessary here.